src/test/java/GoServletTest.java

```diff
@@ -0,0 +1,254 @@
+  private static final String RESTAURANT_ONE = "restaurant near coordinate one";
+  private static final String RESTAURANT_TWO = "restaurant near coordinate two";
+  private static final String BANK_ONE = "bank near coordinate one";
+  private static final String BANK_TWO = "bank near coordinate two";
+
+  private GeocodingResult streetAddressGeocodingResult = new GeocodingResult();
+  private Geometry streetAddressGeometry = new Geometry();
+  private LatLng streetAddressCoordinates = new LatLng(0, 0);
+  private GeocodingResult restaurantGeocodingResult = new GeocodingResult();
+
+  private Duration longerDuration = new Duration();
+  private DirectionsLeg longerLeg = new DirectionsLeg();
+  private DirectionsRoute longerRoute = new DirectionsRoute();
+  private DirectionsResult longerResult = new DirectionsResult();
+
+  private Duration shorterDuration = new Duration();
+  private DirectionsLeg shorterLeg = new DirectionsLeg();
+  private DirectionsRoute shorterRoute = new DirectionsRoute();
+  private DirectionsResult shorterResult = new DirectionsResult();
+
+  @Override
+  @Before
+  public void setUp() throws Exception {
+    super.setUp();
+    directionsClientFactory = Mockito.mock(DirectionsClientFactory.class);
+    directionsClient = Mockito.mock(DirectionsClient.class);
+    placesClientFactory = Mockito.mock(PlacesClientFactory.class);
+    placesClient = Mockito.mock(PlacesClient.class);
+    tasksClientFactory = Mockito.mock(TasksClientFactory.class);
+    tasksClient = Mockito.mock(TasksClient.class);
+    geocodingClientFactory = Mockito.mock(GeocodingClientFactory.class);
+    geocodingClient = Mockito.mock(GeocodingClient.class);
+
+    streetAddressGeocodingResult.geometry = streetAddressGeometry;
+    streetAddressGeocodingResult.geometry.location = streetAddressCoordinates;
+    streetAddressGeocodingResult.types = new AddressType[] {AddressType.STREET_ADDRESS};
+    restaurantGeocodingResult.types = new AddressType[] {AddressType.RESTAURANT};
+
+    longerDuration.inSeconds = 10;
+    longerLeg.duration = longerDuration;
+    longerRoute.legs = new DirectionsLeg[] {longerLeg};
+    longerResult.routes = new DirectionsRoute[] {longerRoute};
+
+    shorterDuration.inSeconds = 5;
+    shorterLeg.duration = shorterDuration;
+    shorterRoute.legs = new DirectionsLeg[] {shorterLeg};
+    shorterResult.routes = new DirectionsRoute[] {shorterRoute};
+
+    servlet =
+        new GoServlet(
+            directionsClientFactory,
+            placesClientFactory,
+            tasksClientFactory,
+            geocodingClientFactory,
+            API_KEY);
+
+    Mockito.when(directionsClientFactory.getDirectionsClient(API_KEY)).thenReturn(directionsClient);
+    Mockito.when(placesClientFactory.getPlacesClient(API_KEY)).thenReturn(placesClient);
+    Mockito.when(tasksClientFactory.getTasksClient(Mockito.any())).thenReturn(tasksClient);
+    Mockito.when(authenticationVerifier.verifyUserToken(Mockito.anyString()))
```

Comment:
Shouldn't have to specify the mock for `authenticationVerfier` - this should be handled by the super class

Reply:
Oh yeah you're right, thanks!

Reply:
`RESOLVED`